File: C++/프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> solution(vector<int> numbers) {
    int N = numbers.size();
    vector<int> answer(N, -1);
    
    for(int i = N - 1; i >= 0; i--) {
        for(int j = i + 1; j < N; j++) {
            if(numbers[i] < numbers[j]) {
                answer[i] = numbers[j];
                break;
            } else if (numbers[i] < answer[j]) {
                answer[i] = answer[j];
                break;
            } else if (answer[j] == -1) {
                break;
            }
        }
    }
    
    return answer;
}

File: Python3/프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.py
def solution(numbers):
    N = len(numbers)
    answer = [-1 for i in range(N)]
    
    for i in range(N - 1, -1, -1):
        for j in range(i + 1, N):
            if numbers[i] < numbers[j]:
                answer[i] = numbers[j]
                break
            elif numbers[i] >= numbers[j] and numbers[i] < answer[j]:
                answer[i] = answer[j]
                break
            elif numbers[i] >= numbers[j] and answer[j] == -1:
                break
        # print(answer)
    return answer